### ISSUE TITLE
`git-annex` fails to compile after upgrading `process`

### ISSUE THREAD

**Issue by hololeap** [gentoo-haskell/gentoo-haskell#1381]
This is related to https://github.com/gentoo-haskell/gentoo-haskell/issues/1377, but trickier since it reveals that multiple packages are still using the old `process`, but there is no automated way to find them, yet.

- `dev-vcs/git-annex-10.20220624`:
  ```
  Warning:
      This package indirectly depends on multiple versions of the same package. This is very likely to cause a compile failure.
        package yesod-static (yesod-static-1.6.1.0-DEY7fLhZIV3792HU5ub76d) requires process-1.6.13.2
        package x509-system (x509-system-1.6.6-Ca9ZSJK06HvLHrunoGAPEV) requires process-1.6.13.2
        package unliftio (unliftio-0.2.23.0-G6aj9zX5QxKLQZBlKGUlqV) requires process-1.6.13.2
        package typed-process (typed-process-0.2.10.1-5SntQ8a4ozw7wD35LUJagh) requires process-1.6.13.2
        package streaming-commons (streaming-commons-0.2.2.4-CwlmgPlCgirCua0nDZqFAe) requires process-1.6.13.2
        package shakespeare (shakespeare-2.0.30-8moO1hPvpvqbFjbKREbpE) requires process-1.6.13.2
        package optparse-applicative (optparse-applicative-0.17.0.0-3NCo4O2c6z1DamS7jVilpD) requires process-1.6.13.2
        package conduit-extra (conduit-extra-1.3.6-HuO6sisAyV38ACqZyhTLPj) requires process-1.6.13.2
        package concurrent-output (concurrent-output-1.10.12-2umcFgjWWbqFWH0YI7bwRx) requires process-1.6.13.2
        package git-annex (git-annex-10.20220624) requires process-1.6.16.0-ACvEVjiix4iDn7Ym1NVURI
  ```
  
  ```
  [482 of 676] Compiling Test.Framework   ( Test/Framework.hs, dist/build/git-annex/git-annex-tmp/Test/Framework.dyn_o )
  
  Test/Framework.hs:770:67: error:
      • Couldn't match expected type ‘process-1.6.13.2:System.Process.Common.CreateProcess’
                    with actual type ‘CreateProcess’
        NB: ‘process-1.6.13.2:System.Process.Common.CreateProcess’
              is defined in ‘System.Process.Common’ in package ‘process-1.6.13.2’
            ‘CreateProcess’
              is defined in ‘System.Process.Common’ in package ‘process-1.6.16.0’
      • In the first argument of ‘createProcessConcurrent’, namely ‘p’
        In a stmt of a 'do' block:
          (_, _, _, pid) <- createProcessConcurrent p
        In the expression:
          do let subdir = tmpdir </> show n
             ensuredir subdir
             let p = ...
             (_, _, _, pid) <- createProcessConcurrent p
             ....
      |
  770 |                         (_, _, _, pid) <- createProcessConcurrent p
      |                                                                   ^
  
  Test/Framework.hs:771:40: error:
      • Couldn't match expected type ‘ProcessHandle’
                    with actual type ‘process-1.6.13.2:System.Process.Common.ProcessHandle’
        NB: ‘ProcessHandle’
              is defined in ‘System.Process.Common’ in package ‘process-1.6.16.0’
            ‘process-1.6.13.2:System.Process.Common.ProcessHandle’
              is defined in ‘System.Process.Common’ in package ‘process-1.6.13.2’
      • In the first argument of ‘waitForProcess’, namely ‘pid’
        In a stmt of a 'do' block: waitForProcess pid
        In the expression:
          do let subdir = tmpdir </> show n
             ensuredir subdir
             let p = ...
             (_, _, _, pid) <- createProcessConcurrent p
             ....
      |
  771 |                         waitForProcess pid
      |                                        ^^^
  ```

**Comment by hololeap** [comment]
Reported by `ForestGuy` on `#gentoo-haskell`

**Comment by hololeap** [comment]
This is hacky but will probably work:

```bash
until find /usr/lib64/ghc-9.0.2/package.conf.d/ -name '*.conf' -print0 \
	| xargs -0 grep -l process-1.6.13.2 \
	| xargs qfile -q \
	| grep -v 'dev-lang/ghc' \
	| xargs -d\\n emerge --deep --oneshot --keep-going
do
	haskell-updater -- @preserved-rebuild
done
```

This needs to be integrated into `haskell-updater`, somehow.

**Comment by hololeap** [comment]
A simple fix would be to go through the packages reported by the command above and make sure they have `dev-haskell/process:=` in `RDPEND`, then revbump them.

**Comment by apraga** [comment]
> This is hacky but will probably work:
> 
> ```shell
> until find '/usr/lib64/ghc-9.0.2/package.conf.d/' -name '*.conf' -print0 \
> 	| xargs -0 grep -l 'process-1.6.13.2' \
> 	| xargs -d\\n qfile -q \
> 	| grep -v 'dev-lang/ghc' \
> 	| xargs -d\\n emerge --deep --oneshot --keep-going
> do
> 	haskell-updater -- @preserved-rebuild
> done
> ```
> 
> This needs to be integrated into `haskell-updater`, somehow.

This solved my issue with git-annex. Thanks !

**Comment by hololeap** [comment]
This should be fixed with fb9738bbe1df2cf0900c1c5693c5ac6f7aa0db08.
